### PR TITLE
Created FetchProduct service

### PR DESCRIPTION
**Pull request by kevinoshea**

### CODE REVIEW

File: frontend/src/lib/services/FetchProducts.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  }
+  if (responseData.some(element => element._id === undefined)) {
+    return false;
+  }
+  if (responseData.some(element => element.favourite === undefined)) {
+    return false;
+  }
+  if (
+    responseData.some(
+      element => element.builds === undefined || !Array.isArray(element.builds) || element.builds.some(build => typeof build !== 'string')
```
Comment (ryan-rushton) [inline frontend/src/lib/services/FetchProducts.ts]: Probs don't need the undefined check here the array one will do it. 

I am thinking these things are going to be used over and over while deserialising responses. Might be work extracting them to some util functions at some point.

Something like 
```
getArrayOfStringsChecker = (attr: string) => (element: any): boolean => !Array.isArray(element[attr]) || element[attr].some(thing => typeof thing !== 'string')
```

File: frontend/src/lib/rest/Rest.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ const handleError = (error, url, message) => {
   showNotification({ message, severity: 'error' });
 };
 
-const doGetRequest = (path, errorMessage = 'rest.error.get') => {
+export const doGetRequest = (path, errorMessage = 'rest.error.get', isExpectedResponse, onSuccess) => {
```
[the remark this answers is not in the document]
Comment (kevinoshea) [reply frontend/src/lib/rest/Rest.js]: Agreed, I think doing the next one will reveal what can be commonised and if this is the right pattern going forward.

File: frontend/src/lib/services/FetchProducts.ts (hunk 1 above)
Comment (kevinoshea) [reply frontend/src/lib/services/FetchProducts.ts]: Ah yep I'll remove the undefined check. Yeah I'm not 100% happy with this atm, the main two issues are the boilerplate stuff and the fact that if it fails you don't know which part failed. I'll try to come up with something for these in the next one.

File: frontend/src/lib/services/FetchProducts.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    products[productName].versionList.push(createVersion(element));
+    products[productName].versionList.sort(versionComparator);
+  });
+  const productList = Object.values(products);
+  productList.sort((a: Product, b: Product): number => a.name.localeCompare(b.name));
+  return productList;
+};
+
+const fetchProducts = (): Promise<Product[]> => {
+  return doGetRequest('/report', 'rest.error.summaryOfReports', isExpectedResponse, (responseData: ResponseData) => {
```
[the remark this answers is not in the document]
Comment (kevinoshea) [reply frontend/src/lib/services/FetchProducts.ts]: Yeah sounds good, I expect we'll need to revisit these a couple of times before we get it right. Still not 100% sure about the onSuccess param and the then/catch stuff.

**Pull request merged**